pkg/workloads/console/integration/integration_test.go

```diff
@@ -66,37 +88,211 @@ var _ = Describe("Console", func() {
 
 	Describe("Creating resources", func() {
 		It("Sets console.spec.user from rbac", func() {
+			consoleTemplate := &workloadsv1alpha1.ConsoleTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-first-console-template",
+					Namespace: namespace,
+				},
+				Spec: workloadsv1alpha1.ConsoleTemplateSpec{
+					AdditionalAttachSubjects: []rbacv1.Subject{},
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								corev1.Container{
+									Image:   "alpine:latest",
+									Name:    "console-container-0",
+									Command: []string{"sleep", "100"},
+								},
+							},
+							RestartPolicy: "Never",
+						},
+					},
+				},
+			}
+
+			By("Creating console template")
+			Expect(mgr.GetClient().Create(context.TODO(), consoleTemplate)).NotTo(
+				HaveOccurred(), "failed to create Console Template",
+			)
+
 			csl := &workloadsv1alpha1.Console{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      "foo",
+					Name:      "my-first-console",
 					Namespace: namespace,
 				},
 				Spec: workloadsv1alpha1.ConsoleSpec{
-					User: "", // deliberately not configured, this should be set by the webhook
+					ConsoleTemplateRef: corev1.LocalObjectReference{Name: "my-first-console-template"},
+					User:               "", // deliberately not configured, this should be set by the webhook
 				},
 			}
 
-			By("Creating Console 'foo'")
+			By("Creating console")
 			Expect(mgr.GetClient().Create(context.TODO(), csl)).NotTo(
-				HaveOccurred(), "failed to create 'foo' Console",
+				HaveOccurred(), "failed to create Console",
 			)
 
 			By("Expect webhook was invoked")
-			Eventually(whcalls, timeout).Should(
+			Eventually(mutWhCalls, timeout).Should(
 				Receive(
-					integration.HandleResource(namespace, "foo"),
+					integration.HandleResource(namespace, "my-first-console"),
 				),
 			)
 
 			By("Expect reconcile succeeded")
 			Eventually(calls, timeout).Should(
 				Receive(
-					integration.ReconcileResourceSuccess(namespace, "foo"),
+					integration.ReconcileResourceSuccess(namespace, "my-first-console"),
 				),
 			)
 
 			By("Expect console.spec.user to be set")
 			Expect(csl.Spec.User).To(Equal("system:unsecured"))
 		})
+
+		It("Denies requests with a console template ref pointing to a nonexistent template", func() {
```

Comment:
I think we'll want to nix this, given a convergent validation strategy. Going to skip past it for now.